prow/cmd/checkconfig/main.go

```diff
@@ -396,6 +409,34 @@ func (c *orgRepoConfig) intersection(c2 *orgRepoConfig) *orgRepoConfig {
 	return res
 }
 
+// union returns a new orgRepoConfig that represents the set union of the
+// repos specified by the receiver and the parameter orgRepoConfigs
+func (c *orgRepoConfig) union(c2 *orgRepoConfig) *orgRepoConfig {
+	res := &orgRepoConfig{
+		orgExceptions: make(map[string]sets.String),
+		repos:         sets.NewString(),
+	}
+
+	for org, excepts1 := range c.orgExceptions {
+		// keep only items in both blacklists
+		if excepts2, ok := c2.orgExceptions[org]; ok {
+			res.orgExceptions[org] = excepts1.Intersection(excepts2)
+		} else {
+			res.orgExceptions[org] = excepts1
+		}
+	}
+
+	for org, excepts2 := range c2.orgExceptions {
+		// update any blacklists not previously updated
+		if _, exists := res.orgExceptions[org]; !exists {
+			res.orgExceptions[org] = excepts2
+		}
+	}
+
+	res.repos = c.repos.Union(c2.repos)
```

Comment:
This isn't quite right. Consider the union of the orgRepoConfigs for the following two queries:
- `org: foo -repo:foo/bar`
- `repo:foo/bar`

This code would output an orgRepoConfig corresponding to: `org:foo -repo:foo/bar repo:foo/bar`.

Please add tests for this function.